### PR TITLE
Configurable resources for git-sync sidecar

### LINKED ISSUE #16056: [Helm] Resources for the git-sync sidecar
**Description**

It would be nice to be able to specify resources for the `git-sync` sidecar in the helm chart values.

**Use case / motivation**

I don't want to use keda for autoscaling and would like to setup a HPA myself. However this is currently not possible since it is not possible to specify resources for the `git-sync` sidecar.

**Are you willing to submit a PR?**

Yes, I am willing to submit a PR.

**Related Issues**

Not that I know of.

### PR DESCRIPTION
**Pull request by CyberHippo**
Add configurable resources for the `git-sync` sidecar.

closes: #16056

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
Fantastic! LGTM

File: chart/tests/test_git_sync_webserver.py [hunk 1]
```diff
@@ -148,3 +148,29 @@ def test_should_add_env(self):
...
+        )
+        assert "128Mi" == jmespath.search(
+            "spec.template.spec.containers[1].resources.limits.memory", docs[0]
+        )
+        assert "169Mi" == jmespath.search(
+            "spec.template.spec.containers[1].resources.requests.memory", docs[0]
+        )
+        assert "300m" == jmespath.search(
+            "spec.template.spec.containers[1].resources.requests.cpu", docs[0]
+        )
```
Comment (kaxil) [inline chart/tests/test_git_sync_webserver.py]:
```suggestion
        assert "300m" == jmespath.search("spec.template.spec.containers[1].resources.requests.cpu", docs[0])
```

File: chart/tests/test_git_sync_webserver.py [hunk 2]
```diff
@@ -148,3 +148,29 @@ def test_should_add_env(self):
...
+                            "requests": {"cpu": "300m", 'memory': "169Mi"},
+                        },
+                    },
+                },
+            },
+            show_only=["templates/webserver/webserver-deployment.yaml"],
+        )
+        assert "128Mi" == jmespath.search(
+            "spec.template.spec.containers[1].resources.limits.memory", docs[0]
+        )
```
Comment (kaxil) [inline chart/tests/test_git_sync_webserver.py]:
```suggestion
        assert "128Mi" == jmespath.search("spec.template.spec.containers[1].resources.limits.memory", docs[0])
```

**Review by kaxil — APPROVED** [review]

**Comment by potiuk** [comment]
Only failed for mssql because of no-rebase. Merging anyway.

**Pull request merged**